components/core/src/clp/clo/clo.cpp

```diff
@@ -47,6 +57,156 @@ enum class SearchFilesResult {
+    }
+
+    try {
+        // Create output directory in case it doesn't exist
+        auto output_dir = boost::filesystem::path(command_line_args.get_ir_output_dir());
+        if (auto error_code = create_directory(output_dir.parent_path().string(), 0700, true);
+            ErrorCode_Success != error_code)
+        {
+            SPDLOG_ERROR(
+                    "Failed to create {} - {}",
+                    output_dir.parent_path().c_str(),
+                    strerror(errno)
+            );
+            return false;
+        }
+
+        Archive archive_reader;
+        archive_reader.open(archive_path.string());
+        archive_reader.refresh_dictionaries();
+
+        auto const& file_split_id = command_line_args.get_file_split_id();
+        auto file_metadata_ix_ptr = archive_reader.get_file_iterator_by_split_id(file_split_id);
+        if (false == file_metadata_ix_ptr->has_next()) {
+            SPDLOG_ERROR(
+                    "File split doesn't exist {} in the archive {}",
+                    file_split_id,
+                    archive_path.string()
+            );
+            return false;
+        }
+
+        mongocxx::client client;
+        mongocxx::collection collection;
+        std::vector<bsoncxx::document::value> results;
+
+        try {
+            auto mongo_uri = mongocxx::uri(command_line_args.get_ir_mongodb_uri());
+            client = mongocxx::client(mongo_uri);
+            collection
+                    = client[mongo_uri.database()][command_line_args.get_ir_mongodb_collection()];
+        } catch (mongocxx::exception const& e) {
+            throw CloOperationFailed(ErrorCode_BadParam_DB_URI, __FILE__, __LINE__);
+        }
+
+        auto ir_output_handler = [&](boost::filesystem::path const& src_ir_path,
+                                     string const& orig_file_id,
+                                     size_t begin_message_ix,
+                                     size_t end_message_ix,
+                                     bool is_last_ir_chunk) {
+            auto dest_ir_file_name = orig_file_id;
+            dest_ir_file_name += "_" + std::to_string(begin_message_ix);
+            dest_ir_file_name += "_" + std::to_string(end_message_ix);
+            dest_ir_file_name += cIrFileExtension;
+
+            auto const dest_ir_path = output_dir / dest_ir_file_name;
+            try {
+                boost::filesystem::rename(src_ir_path, dest_ir_path);
+            } catch (boost::filesystem::filesystem_error const& e) {
+                SPDLOG_ERROR(
+                        "Failed to rename from {} to {}. Error: {}",
+                        src_ir_path.c_str(),
+                        dest_ir_path.c_str(),
+                        e.what()
+                );
+                return false;
+            }
+            results.emplace_back(std::move(bsoncxx::builder::basic::make_document(
+                    bsoncxx::builder::basic::kvp("ir_path", dest_ir_path.string()),
```

Comment:
```suggestion
                    bsoncxx::builder::basic::kvp("path", dest_ir_path.string()),
```

Since this is going into a collection specific for IR files.